### PR TITLE
feat: Improve separation of the product per store and security handling

### PR DESCRIPTION
**Pull request by adrien2p**

### CODE REVIEW

File: medusa-config.js [hunk 1]
```diff
@@ -1,45 +1,45 @@
-// CORS when consuming Medusa from admin
-const ADMIN_CORS = process.env.ADMIN_CORS || "http://localhost:7000,http://localhost:7001";
+import * as dotenv from 'dotenv';
```
Comment (shahednasser) [inline medusa-config.js:1]: Are the changes in this file necessary? I think it's ok to keep them as it is installed with Medusa

**Review by shahednasser — COMMENTED** [review]
Hi Adrien, thank you for submitting this PR! It's definitely very helpful.

I have one concern though: If we want to reflect this change into the article (mainly creating additional modules for Auth and store and explaining the code behind it) it might make the article even longer. 

Do you think it's possible to make the change in the current user module? That way we can just make a change in the existing module and middleware in the article rather than adding an additional set of classes.

If you think it's necessary to put them in different modules, then I can go ahead and make the changes to the article. Just wanted to get your opinion on this.

File: medusa-config.js (hunk 1 above)
Comment (adrien2p) [reply medusa-config.js:1]: Medusa only load .env file which does not allow to set different env depending on the execution environment.

Comment (adrien2p) [reply medusa-config.js:1]: Also, the config file is the only place that calls the dotenv package. That config is called in two places during the loading, expressLoader and passport which are both called later during the loading flow. It ensure that at any time the medusa-config is properly populated no matter the environment. If I am missing something, please point it out to me :)

**Comment by adrien2p** [comment]
> Hi Adrien, thank you for submitting this PR! It's definitely very helpful.
> 
> I have one concern though: If we want to reflect this change into the article (mainly creating additional modules for Auth and store and explaining the code behind it) it might make the article even longer.
> 
> Do you think it's possible to make the change in the current user module? That way we can just make a change in the existing module and middleware in the article rather than adding an additional set of classes.
> 
> If you think it's necessary to put them in different modules, then I can go ahead and make the changes to the article. Just wanted to get your opinion on this.

I think that we could keep it under the userModule but I did that because I dont think that it is related to the user. This guard is of an higher level that the userModule but in order to keep things simple I could move it back to the user module ;)

File: src/modules/user/middlewares/loggdeInUser.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { NextFunction, Response } from 'express';
+import UserService from '../../user/services/user.service';
+import * as jwt from 'jsonwebtoken';
+import { JwtPayload } from "jsonwebtoken";
+
+@Middleware({ requireAuth: true, routes: [{ method: "all", path: '*' }] })
+export class LoggedInUserMiddleware implements MedusaMiddleware {
+    public async consume(req: MedusaAuthenticatedRequest, res: Response, next: NextFunction): Promise<void> {
+        const userService = req.scope.resolve('userService') as UserService;
+        const userData: JwtPayload | string = jwt.decode(req?.session?.jwt);
```
Comment (adrien2p) [inline src/modules/user/middlewares/loggdeInUser.middleware.ts:11]: I wanted to update that part, could you remove everything related to jsonwebtoken and just use req.user.user_id ? It should be available

Comment (shahednasser) [reply src/modules/user/middlewares/loggdeInUser.middleware.ts:11]: Actually, the main issue I ran into while working on the tutorial was that req.user was not available. It always was undefined

File: medusa-config.js (hunk 1 above)
Comment (shahednasser) [reply medusa-config.js:1]: Ok I don't mind making this change but is it necessary to mention in the tutorial? Does it play part in any of the other parts?

**Pull request merged**